lib/iris/tests/__init__.py

```diff
@@ -205,6 +205,14 @@ def setUpClass(cls):
         iris.site_configuration['cf_profile'] = None
 
     def _assert_str_same(self, reference_str, test_str, reference_filename, type_comparison_name='Strings'):
+        # Work around to deal with alterations to attribute typing in
+        # netcdf4python, introduced in 1.1 and present in 1.2. see:
+        # https://github.com/Unidata/netcdf-c/issues/298
+        # https://github.com/Unidata/netcdf4-python/issues/529
+        # https://github.com/Unidata/netcdf4-python/issues/575
+        # amongst others for further details.
+        if type_comparison_name == 'CDL' and reference_str != test_str:
+            test_str = test_str.replace('\t\tstring ', '\t\t')
```

Comment:
@marqh Matching with a replace in this way is kinda open to making incorrect changes.

Perhaps you could be more rigorous and use a compiled regexp that also anchors the pattern match to the beginning of a line with globbed white space ...